libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/navigation/CarNavigationCamera.kt

```diff
@@ -164,7 +146,7 @@ class CarNavigationCamera internal constructor(
         )
 
         mapboxNavigation.registerLocationObserver(locationObserver)
-        routesProvider.registerRoutesListener(routesListener)
+        mapboxNavigation.registerRoutesObserver(routesObserver)
```

Comment:
I don't think observing routes from `MapboxNavigation` is correct here, since routes are not set there during route preview. I believe that's why I can't see the whole route in the preview state with these changes.

Reply:
This is needed for the camera to function in active guidance state. However, I agree that camera is missing observing routes from route preview state. This is how it's been done in the Drop-In UI
```
coroutineScope.launch {
            val routesFlow = mapboxNavigation.flowRoutesUpdated()
                .map { it.navigationRoutes }
                .stateIn(
                    coroutineScope,
                    SharingStarted.WhileSubscribed(),
                    mapboxNavigation.getNavigationRoutes()
                )

            val routePreview = store.select { it.previewRoutes }
            val navigation = store.select { it.navigation }
            combine(
                routePreview,
                routesFlow,
                navigation
            ) { previewRoutes, navigationRoutes, navigationState ->
                val routes = if (previewRoutes is RoutePreviewState.Ready) {
                    previewRoutes.routes
                } else if (navigationRoutes.isNotEmpty()) {
                    navigationRoutes
                } else {
                    emptyList()
                }
                when (routes.isNotEmpty()) {
                    true -> {
                        viewportDataSource.onRouteChanged(routes.first())
                        viewportDataSource.evaluate()
                        when (navigationState) {
                            NavigationState.ActiveNavigation,
                            NavigationState.Arrival -> {
                                store.dispatch(SetCameraMode(TargetCameraMode.Following))
                            }
                            else -> {
                                store.dispatch(SetCameraMode(TargetCameraMode.Overview))
                            }
                        }
                    }
                    false -> {
                        viewportDataSource.clearRouteData()
                        viewportDataSource.evaluate()
                    }
                }
            }.collect()
        }
```
It always looks at the route from both routePreview and onRoutesChanged. Whenever a route is available, the camera viewport updates.

Reply:
Yeah you're right, camera needs to share a common route provider. 

The more generic `RoutesProvider` feels better because of this 🤔